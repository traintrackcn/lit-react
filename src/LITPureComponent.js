import React, { PureComponent } from 'react';

export default class LITPureComponent extends PureComponent {

    constructor(props) {
        super(props);
        this.state = {};
    }

    get_unmounted(){
        return this.unmounted;
    }

    componentWillUnmount(){
        this.unmounted = true;
    }

    componentWillMount(){
        this.onReceiveProps(this.props);
    }

    tryComponentWillMount(){
        
    }

    componentWillReceiveProps(nextProps){
        this.onReceiveProps(nextProps);
    }

    onReceiveProps(props){
        try{
            return this.tryOnReceiveProps(props);
        }catch(e){
            return null;
        }
    }

    tryOnReceiveProps(props){

    }

    render(){
        try{
            return this.tryRender();
        }catch(e){
            return null;
        }
    }

    tryRender(){
        return null;
    }

}