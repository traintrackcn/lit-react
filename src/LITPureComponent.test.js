import LITPureComponent from './LITPureComponent';
import renderer from 'react-test-renderer';
import React from 'react';

it('test LITPureComponent', ()=> {
 
    const tree = renderer.create(<LITPureComponent />).toJSON();
    expect(tree).toMatchSnapshot();
});


it('test child LITPureComponent', () => {

    class ChildComponent extends LITPureComponent{
        constructor(){
            super();
            console.log('_mounted ->', this._mounted);
        }
    }

    new ChildComponent();
});



