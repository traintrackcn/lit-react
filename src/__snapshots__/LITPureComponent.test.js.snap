// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test LITPureComponent 1`] = `null`;
